monkey/monkey_island/cc/resources/agent_configuration.py

```diff
@@ -20,17 +19,17 @@ def __init__(self, agent_configuration_repository: IAgentConfigurationRepository
     # Used by the agent. Can't secure
     def get(self):
         configuration = self._agent_configuration_repository.get_configuration()
-        configuration_json = AgentConfigurationObject.to_json(configuration)
-        return make_response(configuration_json, 200)
+        configuration_dict = configuration.dict()
+        return make_response(configuration_dict, 200)
 
     @jwt_required
     def put(self):
         try:
-            configuration_object = AgentConfigurationObject.from_mapping(request.json)
+            configuration_object = AgentConfigurationObject(**request.json)
             self._agent_configuration_repository.store_configuration(configuration_object)
             # API Spec: Should return 204 (NO CONTENT)
             return make_response({}, 200)
-        except (InvalidConfigurationError, json.JSONDecodeError) as err:
+        except (ValueError, TypeError, json.JSONDecodeError) as err:
```

Comment:
Are `ValueError` and `TypeError` the only errors that can get raised? Would it make sense to instead throw a custom error, since we're already hooking the error handling? If we only throw one, it would simplify the error handling

Reply:
`ValueError` and `TypeError` are for different things. We shouldn't create a custom error that encompasses both.

Reply:
Yes, check the [comment in common/base_models.py](https://github.com/guardicore/monkey/blob/4cbefedad2f575432d3e880991d237471de45ce3/monkey/common/base_models.py#L21..L29).